tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+    """
+
+    def setUp(self):
+        self.doPlot = False  # don't make plots unless specifically requested
+        self.matchRadius = 0.1*geom.arcseconds  # match sources within 0.1" for RMS statistics
+
+    def tearDown(self):
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.jointcalTask
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.catalogs
+
+    def _prep_reference_loader(self, center, radius):
+        """
+        Setup an astrometry.net reference loader.
+
+        @param center (afw.coord) The center of the field you're testing on.
+        @param radius (afw.geom.angle) The radius to load objects around center.
+        """
+        refLoader = LoadAstrometryNetObjectsTask(LoadAstrometryNetObjectsConfig())
+        # Make a copy of the reference catalog for in-memory contiguity.
+        self.reference = refLoader.loadSkyCircle(center, radius, filterName='r').refCat.copy()
+
+    def _testJointCalTask_run(self, nCatalogs, relative_error, absolute_error):
+        """Test jointcal.run() on nCatalogs, requiring less than some error (arcsec)."""
+
+        result = self.jointcalTask.run(self.catalogs[:nCatalogs])
+        self.dataRefs = result.dataRefs
+        self.old_wcss = result.old_wcss
+
+        rms_rel, rms_abs = self.compute_rms(self.catalogs[:nCatalogs], fluxlimit=100)
+        self.assertLess(rms_rel, relative_error/arcsec_per_radian)
+        self.assertLess(rms_abs, absolute_error)
+
+    def _testJointCalTask(self, nCatalogs, relative_error, absolute_error):
+        """Test parseAndRun for jointcal on nCatalogs, requiring less than some error (arcsec)."""
+
+        visits = '^'.join(str(v) for v in self.visitList[:nCatalogs])
+        output_dir = os.path.join('.test', self.__class__.__name__)
+        result = jointcal.JointcalTask.parseAndRun(args=[self.input_dir, '--output', output_dir,
+                                                         '--clobber-versions', '--clobber-config',
+                                                         '--doraise',
+                                                         '--id', 'visit=%s'%visits],
+                                                   doReturnResults=True)
+        self.dataRefs = result.resultList[0].result.dataRefs
+        self.old_wcss = result.resultList[0].result.old_wcss
+
+        rms_rel, rms_abs = self.compute_rms(self.dataRefs, fluxlimit=100)
+        self.assertLess(rms_rel, relative_error/arcsec_per_radian)
+        self.assertLess(rms_abs, absolute_error)
+
+    def compute_rms(self, dataRefs, fluxlimit=100):
+        """
+        Match all dataRefs to compute the RMS, for all detections above some flux limit.
+
+        @param dataRefs (list) the dataRefs to do the relative matching between
+        @param fluxlimit (float) SN (flux/fluxSigma) for sources to be included in match.
+
+        @return (float, float) relative, absolute RMS of stars, in radians
```

Comment:
Amending my comment: all of the RMS calculations now use Astropy Quantities, because those behave well with matplotlib, whereas afw Angles do not.